### ISSUE TITLE
Setting resourceId to newly created event record before it is saved by the event editor fails

### ISSUE THREAD

**Issue by pmiklashevich** [bryntum/support#2056]
https://www.bryntum.com/forum/viewtopic.php?f=44&t=16152

http://lh/bryntum-suite/scheduler/examples/basic/
```
scheduler.on('beforeEventAdd', ({ eventRecord, resourceRecords }) => {
    eventRecord.resourceId = resourceRecords[0].id;
});
```
Drag create an event, save it.
```
EventEdit.js:897 Uncaught (in promise) TypeError: Cannot read property 'id' of undefined
    at EventEdit.finalizeEventSave (EventEdit.js:897)
    at Object.finalize (EventEdit.js:974)
    at EventEdit.js:1006
    at new Promise (<anonymous>)
    at EventEdit.save (EventEdit.js:943)
    at EventEdit.onSaveClick (EditBase.js:674)
    at Button.trigger (Events.js:1122)
    at Button.onInternalClick (Button.js:449)
    at HTMLButtonElement.handler (EventHelper.js:512)
```

The same happens here when 'resourceId' is used as a field name.
https://www.bryntum.com/forum/viewtopic.php?p=80320#p80320

http://lh/bryntum-suite/scheduler/examples/basic/
```
const scheduler = new Scheduler({
    features : {
        eventEdit : {
            editorConfig : {
                items : {
                    resourceField : {
                        name : 'resourceId' // 'resource' is used by default
                    }
                }
            }
        }
    },
```
Go to browser and create a new event. Try to save.
```
EventEdit.js:897 Uncaught (in promise) TypeError: Cannot read property 'id' of undefined
    at EventEdit.finalizeEventSave (EventEdit.js:897)
    at Object.finalize (EventEdit.js:974)
    at EventEdit.js:1006
    at new Promise (<anonymous>)
    at EventEdit.save (EventEdit.js:943)
    at EventEdit.onSaveClick (EditBase.js:674)
    at Button.trigger (Events.js:1120)
    at Button.onInternalClick (Button.js:449)
    at HTMLButtonElement.handler (EventHelper.js:512)
```

In EventEdit.finalizeEventSave:
```
eventStore.add(eventRecord);
const [assignmentRecord] = eventStore.assignEventToResource(eventRecord, resourceRecords);
```
When 'resourceId' is already set to the eventRecord, eventStore.add adds an event and creates an assignment for it automatically. assignEventToResource does nothing and returns an empty array in this case. So assignmentRecord is null. Extracting assignments explicitly works:
```
eventStore.add(eventRecord);
eventStore.assignEventToResource(eventRecord, resourceRecords);
const [assignmentRecord] = eventRecord.assignments;
```

**Comment by pmiklashevich** [comment]
Hiding our field and creating a custom field with 'resourceId' name is not supported either. See the same error.
```
const resourceStore = new ResourceStore({
    data : resources
});

const scheduler = new Scheduler({
    features         : {
        eventEdit : {
            editorConfig : {
                items : {
                    resourceField   : false,
                    myResourceField : {
                        weight       : 210, // resourceField weight is 200
                        type         : 'combo',
                        label        : 'Employees',
                        valueField   : 'id',
                        displayField : 'name',
                        name         : 'resourceId',
                        store        : resourceStore.chain()
                    }
                }
            }
        }
    },
    appendTo         : 'container',
    minHeight        : '20em',
    resourceStore    : resourceStore,
    events           : events,
```

**Comment by pmiklashevich** [comment]
The issue was fixed by adding this code in #2364:
```
//  First assign the resource, then add to eventStore to not fire extra 'update' event
const [assignmentRecord] = eventStore.assignEventToResource(eventRecord, resourceRecords);
eventStore.add(eventRecord);
```
'Should be possible to set resource id in beforeEventAdd handler' - fixed
'Should be possible to change resource field name' - fixed
'Setting resourceField to false and using another field with custom ref' - not supported. "resourceField" reference must be used. It's possible to override the resource field config if needed.

Tests are added in the PR
